code/z_adventurezones/space_casino.dm

```diff
@@ -0,0 +1,358 @@
+/*
+    Lythine's space casino prefab
+	Contents:
+	 Item slot machine
+	 Barbuddy
+	 Misc props
+*/
+
+// Item slot machine
+
+/obj/submachine/slot_machine/item
+	name = "Item Slot Machine"
+	desc = "A slot machine that produces items rather than money. Somehow."
+	icon_state = "slotsitem-off"
+	mats = 50
+	var/uses = 0
+
+	var/list/junktier = list( // junk tier, 60% chance
+		"/obj/item/a_gift/easter",
+		"/obj/item/raw_material/rock",
+		"/obj/item/balloon_animal",
+		"/obj/item/cigpacket",
+		"/obj/item/clothing/shoes/moon",
+		"/obj/item/fish/carp",
+		"/obj/item/instrument/bagpipe",
+		"/obj/item/clothing/under/gimmick/yay",
+		"/obj/item/scrap",
+		"/obj/item/paper_bin",
+		"/obj/item/item_box/stickers",
+		"/obj/item/storage/box/costume/hotdog",
+		"/mob/living/critter/small_animal/cockroach",
+		"/obj/item/device/light/flashlight",
+		"/obj/item/kitchen/utensil/knife",
+		"/obj/item/staple_gun",
+		"/obj/item/old_grenade/banana/cheese_sandwich",
+		"/obj/item/old_grenade/banana/banana_corndog",
+		"/obj/item/rubberduck"
+	)
+
+	var/list/usefultier = list( // half decent tier, 35% chance
+		"/obj/item/clothing/gloves/yellow",
+		"/obj/item/bat",
+		"/obj/item/reagent_containers/food/snacks/donkpocket/warm",
+		"/obj/item/device/flash",
+		"/obj/item/clothing/glasses/sunglasses",
+		"/obj/vehicle/skateboard",
+		"/obj/item/storage/firstaid/regular",
+		"/obj/item/clothing/shoes/sandal",
+		"/obj/item/cigpacket/random",
+		"/obj/item/clothing/mask/gas",
+		"/obj/critter/domestic_bee/trauma",
+		"/obj/item/storage/firstaid/crit"
+	)
+
+	var/list/raretier = list( // rare tier, 5%~ chance
+		"/obj/item/hand_tele",
+		"/obj/item/baton",
+		"/obj/item/clothing/suit/armor/vest",
+		"/obj/item/device/voltron",
+		"/obj/item/gun/energy/phaser_gun",
+		"/obj/item/gimmickbomb/hotdog",
+		"/obj/item/card/id/captains_spare",
+		"/obj/item/storage/banana_grenade_pouch",
+		"/mob/living/critter/wendigo", // have fun!
+		"/obj/item/artifact/teleport_wand",
+		"/obj/machinery/vehicle/tank/car/blue", // A BRAAAAAAND NEEEEEEW CAAAAAAAAAR!
+		"/obj/item/card/id/dabbing_license",
+		"/obj/item/gun/energy/egun"
+	)
+
+	var/list/veryraretier = list( // very rare tier, 0.2% chance
+		"/obj/item/sword_core",
+		"/obj/item/storage/bowling",
+		"/obj/item/storage/belt/wrestling",
+		"/obj/item/stimpack",
+		"/obj/item/clothing/head/bighat/syndicate",
+		"/obj/item/gun/energy/laser_gun/pred",
+		"/obj/item/reagent_containers/emergency_injector/high_capacity/donk_injector",
+		"/obj/item/clothing/suit/space/industrial/syndicate"
+	)
```

Comment:
Ehhh...

Reply:
I'm a bit uncertain of what should be considered too much to receive from the machine. Currently planning on removing the very rare tier entirely and removing a few items from the rare tier (namely the guns and the baton) and making the chances of rolling an item from the rare tier somewhat smaller.